tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/functions/CustomFunctionTest.java

```diff
@@ -42,4 +42,23 @@ public void testDuplicateParameters() {
         // Checking duplicate parameter definition in a function starting at 35st column
         BAssertUtil.validateError(compile, 0, "redeclared symbol 'param'", 1, 35);
     }
+
+    @Test
+    public void testAssignValueToFunctionNegative() {
+        CompileResult compile = BCompileUtil.compile("test-src/functions/assign_value_to_function_negative.bal");
+        int i = 0;
+        BAssertUtil.validateError(compile, i++, "invalid assignment: 'function' declaration is final", 20, 5);
+        BAssertUtil.validateError(compile, i++, "invalid assignment: 'function' declaration is final", 21, 5);
+        Assert.assertEquals(compile.getErrorCount(), i);
+    }
+
+    @Test
+    public void testAssignValueToObjectMethodNegative() {
+        CompileResult compile = BCompileUtil.compile("test-src/functions/assign_value_to_object_method.bal");
+        int i = 0;
+        BAssertUtil.validateError(compile, i++, "cannot update 'final' object field '$anonType$$myObj$$_0.getVal'",
```

Comment:
Make use of a type reference and give a name to the anon type

Reply:
Updated error msgs 2071b739b73103f9352bc9c60afc5320df799740